handler/src/main/java/io/netty/handler/ssl/OpenSslCachingKeyMaterialProvider.java

```diff
@@ -28,10 +28,13 @@
  */
 final class OpenSslCachingKeyMaterialProvider extends OpenSslKeyMaterialProvider {
 
+    private final int maxCachedEntries;
+    private volatile boolean full;
```

Comment:
I think non-volatile would also be fine/potentially cheaper for this?

Reply:
imho volatile is "more correct" and I am not worried about the perf hit.